### PR TITLE
Complete fetch tutorial, fix navbar typo in projects.html

### PR DESCRIPTION
**Pull request by SylviaZiyuZhang**
Implemented book recommendation with fetch at /literature

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,30 @@
...
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+@WebServlet("/literature")
+public final class DataServlet extends HttpServlet {
```
Comment (ThomasQuesadilla) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: Is the final necessary?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,30 @@
...
 
+  private List<String> BookRecs;
+  @Override
+  public void init() {
+    BookRecs = new ArrayList<>();
+    BookRecs.add("The Thread by Victoria Hislop");
+    BookRecs.add("The Hunchback of Notre Dame by Victor Hugo");
+    BookRecs.add("The Brothers Karamazov by Fyodor Dostoevsky");
```
Comment (ThomasQuesadilla) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: just a btw, but another way of doing this is 
`new ArrayList<>(Arrays.asList(element 1, element 2,...))`

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (SylviaZiyuZhang) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: In retrospect, it helps prevent accidental overriding as more code comes in, but it's not necessary for the functionality.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -36,4 +36,10 @@ function FloatingNavbar() {
   else {
     navbar.classList.remove("sticky");
   }
+}
+
+function getBookRec() {
+  fetch('/literature').then(response => response.text()).then((quote) => {
```
Comment (linguistrix) [inline portfolio/src/main/webapp/script.js:42]: Ideally you would have some error handling logic here in case you don't get an expected response

Comment (SylviaZiyuZhang) [reply portfolio/src/main/webapp/script.js:42]: Oops, I just followed the example code snippet - I'll look into HTTP error handling!

**Pull request merged**